python-devicetree tox run fails

**Describe the bug**

tox test for `python-devicetree` fails with the following message:

```
py3 run-test: commands[1] | python -m mypy --config-file=/home/runner/work/zephyr/zephyr/scripts/dts/python-devicetree/tox.ini --package=devicetree
src/devicetree/edtlib.py:2951: error: Cannot infer type argument 1 of "add_constructor" of "BaseConstructor"
src/devicetree/edtlib.py:2951: error: Cannot infer type argument 2 of "add_constructor" of "BaseConstructor"
src/devicetree/edtlib.py:2953: error: "object" has no attribute "construct_pairs"
Found 3 errors in 1 file (checked 4 source files)
```

This is broken since the `types-PyYAML` version 6.0.8 (latest as of now); it succeeds with 6.0.7.

https://github.com/zephyrproject-rtos/zephyr/runs/6744542428?check_suite_focus=true#step:8:25

**To Reproduce**

Run `tox` inside `scripts/dts/python-devicetree`.

**Expected behavior**

tox test succeeds.

**Impact**

CI reports a failure.

--- a/scripts/dts/python-devicetree/tox.ini
+++ b/scripts/dts/python-devicetree/tox.ini
@@ -5,7 +5,7 @@ envlist=py3
 deps =
     setuptools-scm
     pytest
-    types-PyYAML
+    types-PyYAML==6.0.7
     mypy
 setenv =
     TOXTEMPDIR={envtmpdir}

